packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -4,219 +4,326 @@
+@property(strong, nonatomic) NSString *path;
 
 @end
 
 @implementation FLTGoogleMapMarkerController
 
 - (instancetype)initMarkerWithPosition:(CLLocationCoordinate2D)position
                             identifier:(NSString *)identifier
-                               mapView:(GMSMapView *)mapView {
-  self = [super init];
-  if (self) {
-    _marker = [GMSMarker markerWithPosition:position];
-    _mapView = mapView;
-    _marker.userData = @[ identifier ];
-  }
-  return self;
+                               mapView:(GMSMapView *)mapView
+                             iconImage:(UIImage *)iconImage
+                              fontPath:(nonnull NSString *)path {
+    self = [super init];
+    if (self) {
+        _marker = [GMSMarker markerWithPosition:position];
+        _mapView = mapView;
+        _marker.userData = @[ identifier ];
+        _iconImage = iconImage;
+        _path = path;
+    }
+    return self;
 }
 
 - (void)showInfoWindow {
-  self.mapView.selectedMarker = self.marker;
+    self.mapView.selectedMarker = self.marker;
 }
 
 - (void)hideInfoWindow {
-  if (self.mapView.selectedMarker == self.marker) {
-    self.mapView.selectedMarker = nil;
-  }
+    if (self.mapView.selectedMarker == self.marker) {
+        self.mapView.selectedMarker = nil;
+    }
 }
 
 - (BOOL)isInfoWindowShown {
-  return self.mapView.selectedMarker == self.marker;
+    return self.mapView.selectedMarker == self.marker;
 }
 
 - (void)removeMarker {
-  self.marker.map = nil;
+    self.marker.map = nil;
 }
 
 - (void)setAlpha:(float)alpha {
-  self.marker.opacity = alpha;
+    self.marker.opacity = alpha;
 }
 
 - (void)setAnchor:(CGPoint)anchor {
-  self.marker.groundAnchor = anchor;
+    self.marker.groundAnchor = anchor;
 }
 
 - (void)setDraggable:(BOOL)draggable {
-  self.marker.draggable = draggable;
+    self.marker.draggable = draggable;
 }
 
 - (void)setFlat:(BOOL)flat {
-  self.marker.flat = flat;
+    self.marker.flat = flat;
 }
 
 - (void)setIcon:(UIImage *)icon {
-  self.marker.icon = icon;
+    self.marker.icon = icon;
 }
 
 - (void)setInfoWindowAnchor:(CGPoint)anchor {
-  self.marker.infoWindowAnchor = anchor;
+    self.marker.infoWindowAnchor = anchor;
 }
 
 - (void)setInfoWindowTitle:(NSString *)title snippet:(NSString *)snippet {
-  self.marker.title = title;
-  self.marker.snippet = snippet;
+    self.marker.title = title;
+    self.marker.snippet = snippet;
 }
 
 - (void)setPosition:(CLLocationCoordinate2D)position {
-  self.marker.position = position;
+    self.marker.position = position;
 }
 
 - (void)setRotation:(CLLocationDegrees)rotation {
-  self.marker.rotation = rotation;
+    self.marker.rotation = rotation;
 }
 
 - (void)setVisible:(BOOL)visible {
-  self.marker.map = visible ? self.mapView : nil;
+    self.marker.map = visible ? self.mapView : nil;
 }
 
 - (void)setZIndex:(int)zIndex {
-  self.marker.zIndex = zIndex;
+    self.marker.zIndex = zIndex;
+}
+
+- (UIImage *)addClusterMarkerText:(NSString *)string {
```

Comment:
Maybe add some newlines between some of the code to make it easier to distinguish each "section"